play/src/front/Components/Cowebsites/JistiCowebsiteComponent.svelte

```diff
@@ -0,0 +1,159 @@
+<script lang="ts">
+    import { onDestroy, onMount } from "svelte";
+    import { get } from "svelte/store";
+    import {
+        JitsiCoWebsite,
+        JitsiApi,
+        defaultInterfaceConfig,
+        mergeConfig,
+        JitsiOptions,
+    } from "../../WebRtc/CoWebsite/JitsiCoWebsite";
+    import { jitsiExternalApiFactory } from "../../WebRtc/JitsiExternalApiFactory";
+    import { screenWakeLock } from "../../Utils/ScreenWakeLock";
+    import { requestedCameraState, requestedMicrophoneState } from "../../Stores/MediaStore";
+    import { currentPlayerWokaStore } from "../../Stores/CurrentPlayerWokaStore";
+    import { userIsJitsiDominantSpeakerStore } from "../../Stores/GameStore";
+    import { inExternalServiceStore } from "../../Stores/MyMediaStore";
+    import { gameManager } from "../../Phaser/Game/GameManager";
+    import { coWebsiteManager } from "../../Stores/CoWebsiteStore";
+
+    export let actualCowebsite: JitsiCoWebsite;
+    let domain = actualCowebsite.getDomain();
+    let jitsiContainer: HTMLDivElement;
+    let roomName: string;
+    let playerName = gameManager.getPlayerName();
+    let jwt: string | undefined;
+    let jitsiApi: JitsiApi;
+    let screenWakeRelease: (() => Promise<void>) | undefined;
+
+    const onDominantSpeakerChanged = (data: { id: string }) => {
+        if (jitsiApi) {
+            userIsJitsiDominantSpeakerStore.set(
+                data.id === actualCowebsite.getCurrentParticipantId(jitsiApi.getParticipantsInfo())
+            );
+        }
+    };
+
+    let onAudioChange = (changeAudio: { muted: boolean }) => {
+        if (changeAudio.muted) {
+            requestedMicrophoneState.disableMicrophone();
+        } else {
+            requestedMicrophoneState.enableMicrophone();
+        }
+    };
+    const onVideoChange = (changeVideo: { muted: boolean }) => {
+        if (changeVideo.muted) {
+            requestedCameraState.disableWebcam();
+        } else {
+            requestedCameraState.enableWebcam();
+        }
+    };
+
+    const onParticipantsCountChange = () => {
+        actualCowebsite.updateParticipantsCountStore();
+    };
+
+    onMount(() => {
+        let cancelled = false;
+
+        inExternalServiceStore.set(true);
+
+        jitsiExternalApiFactory
+            .loadJitsiScript(domain)
+            .then(() => {
+                const options: JitsiOptions = {
+                    roomName: roomName,
+                    jwt: jwt,
+                    width: "100%",
+                    height: "100%",
+                    parentNode: jitsiContainer,
+                    configOverwrite: mergeConfig(),
+                    interfaceConfigOverwrite: {
+                        ...defaultInterfaceConfig,
+                        ...actualCowebsite.jitsiInterfaceConfig,
+                    },
+                };
+
+                const jistiMeetLoadedPromise = new Promise<void>((resolve) => {
```

Comment:
Try to use a `CancelablePromise` here instead and call the "jistiMeetLoadedPromise.cancel()" method in the onDestroy function.